code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -6,113 +6,179 @@
+	var/datum/status_effect/stacking/essence_link/existing_link
+	/// Amount of health restored by this ability.
+	var/heal_amount = 50
+	/// Multiplier applied to this heal when below a certain threshold.
+	var/heal_multiplier = 1
 
 /datum/action/xeno_action/activable/psychic_cure/acidic_salve/use_ability(atom/target)
-	if(owner.do_actions)
+	var/mob/living/carbon/xenomorph/X = owner
+	existing_link = X.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	if(X.do_actions)
 		return FALSE
-
-	if(!do_mob(owner, target, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+	if(!do_mob(X, target, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
 		return FALSE
-
-	owner.visible_message(span_xenowarning("\the [owner] vomits acid over [target]!"), \
-	span_xenowarning("We cover [target] with our rejuvinating goo!"))
-	target.visible_message(span_xenowarning("[target]'s wounds are mended by the acid."), \
-	span_xenowarning("We feel a sudden soothing chill."))
-
-	playsound(target, "alien_drool", 25)
-
-	new /obj/effect/temp_visual/telekinesis(get_turf(target))
-	var/mob/living/carbon/xenomorph/patient = target
-	patient.salve_healing()
-
+	X.visible_message(span_xenowarning("\the [X] vomits acid over [target], mending their wounds!"))
 	owner.changeNext_move(CLICK_CD_RANGE)
-
-	log_combat(owner, patient, "acid salved")	//Not sure how important this log is but leaving it in
-
+	salve_healing(target)
 	succeed_activate()
 	add_cooldown()
 
-/mob/living/carbon/xenomorph/proc/salve_healing() //Slight modification of the heal_wounds proc
-	var/amount = 50	//Smaller than psychic cure, less useful on xenos with large health pools
-	if(recovery_aura)	//Leaving in the recovery aura bonus, not sure if it is too high the way it is
-		amount += recovery_aura * maxHealth * 0.01 // +1% max health per recovery level, up to +5%
-	var/remainder = max(0, amount - getBruteLoss()) //Heal brute first, apply whatever's left to burns
-	adjustBruteLoss(-amount)
-	adjustFireLoss(-remainder, updating_health = TRUE)
-	adjust_sunder(-amount/20)
-
-// ***************************************
-// *********** Drone Jelly
-// ***************************************
-/datum/action/xeno_action/create_jelly/slow
-	cooldown_timer = 45 SECONDS
+/// Heals the target and gives them a regenerative buff, if applicable.
+/datum/action/xeno_action/activable/psychic_cure/acidic_salve/proc/salve_healing(mob/living/carbon/xenomorph/target)
+	var/remaining_health = round(target.maxHealth - (target.getBruteLoss() + target.getFireLoss()))
+	var/health_threshold = round(target.maxHealth / 10) // 10% of the target's maximum health
+	if(existing_link && existing_link.stacks >= 1 && target == existing_link.link_target)
+		target.apply_status_effect(STATUS_EFFECT_XENO_SALVE_REGEN)
+		if(remaining_health <= health_threshold)
+			heal_multiplier = 3
+			playsound(target,'sound/effects/magic.ogg', 75, 1)
+			existing_link.add_stacks(-1)
```

Comment:
Why does this interact with the link??

Reply:
Talked in Discord, will write what's happening with Acidic Salve in the PR.